lib/translator/common.cc

```diff
@@ -54,6 +60,21 @@ void SetAllocPageCallbacks(uint64_t (*alloc_callback)(uint16_t),
   dealloc_pages_callback = dealloc_callback;
 }
 
+uint64_t htonll(uint64_t value) {
+  static uint32_t test_val = 42;
+
+  // Check first byte to determine endianness
+  if (*reinterpret_cast<const char*>(&test_val) == test_val) {  // little endian
```

Comment:
There should be some standard compiler-provded preprocessor macros you can check so that you can use `#ifdef` instead of `if(...)`, but this is probably sufficient for our purposes at this time.